app/models/piece.rb

```diff
@@ -3,6 +3,20 @@ class Piece < ActiveRecord::Base
 
   enum color: [:white, :black]
 
+  def move_to!(new_x, new_y)
+    destination_piece = game.pieces.find_by(x_coordinate: new_x, y_coordinate: new_y)
+    if destination_piece.present?
+      if destination_piece.color == color
+        'Error'
+      else destination_piece.color != color
```

Comment:
this can be just `else`. No need to check the color here, since it's checked on line 9 in the `if`